SofaKernel/modules/SofaBaseLinearSolver/CGLinearSolver.inl

```diff
@@ -175,32 +184,39 @@ void CGLinearSolver<TMatrix,TVector>::solve(Matrix& M, Vector& x, Vector& b)
             }
         }
 
-        if( nb_iter==1 )
-            p = r; //z;
-        else
+        /// Compute the value of p, conjugate with x
+        if( nb_iter==1 )    // FIRST step
+            p = r;
+        else                // ALL other steps
         {
             beta = rho / rho_1;
-            //p = p*beta + r; //z;
+
+            /// Update p = p*beta + r;
             cgstep_beta(params, p,r,beta);
         }
 
+
         if( verbose )
         {
             sout<<"p : "<<p<<sendl;
         }
 
-        // matrix-vector product
+        /// Compute the matrix-vector product : M p
         q = M*p;
```

Comment:
The comments indeed help understanding the solver. I had a bit of trouble with that line when I realized it also triggers the visitor(s) that will gather the necessary data for that particular multiplication. It seems a bit much for a simple "=" and hides a lot of what goes on in Sofa. (This comment is not to fix it here, just to maybe start a discussion about it)